From 449155f08d7cc1cbefef5fed7e7231bca3c56c49 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 17 Apr 2025 17:40:10 +0100
Subject: [PATCH] crypto: fix up OPENSSL_NO_FIPS_JITTER code path

The OPENSSL_NO_FIPS_JITTER codepath is not working correctly, because
the codepaths that rely on such define did not include configuration
header to get the define.

This leads to jitter entropy not being used, when it is expected to be
used.

Fixes: a7c0fa601e (Add ifndef to seed-src_jitter too, 2025-02-23)
---
 crypto/provider_core.c                            | 1 +
 crypto/rand/rand_lib.c                            | 1 +
 providers/fips/fipsprov.c                         | 1 +
 providers/implementations/rands/seed_src_jitter.c | 1 +
 4 files changed, 4 insertions(+)

diff --git a/crypto/provider_core.c b/crypto/provider_core.c
index 490991b5e5..94d23f7852 100644
--- a/crypto/provider_core.c
+++ b/crypto/provider_core.c
@@ -8,6 +8,7 @@
  */
 
 #include <assert.h>
+#include <openssl/configuration.h>
 #include <openssl/core.h>
 #include <openssl/core_dispatch.h>
 #include <openssl/core_names.h>
diff --git a/crypto/rand/rand_lib.c b/crypto/rand/rand_lib.c
index 9233322b5f..c3fc0b4072 100644
--- a/crypto/rand/rand_lib.c
+++ b/crypto/rand/rand_lib.c
@@ -12,6 +12,7 @@
 
 #include <openssl/err.h>
 #include <openssl/opensslconf.h>
+#include <openssl/configuration.h>
 #include <openssl/core_names.h>
 #include <openssl/provider.h>
 #include "internal/cryptlib.h"
diff --git a/providers/fips/fipsprov.c b/providers/fips/fipsprov.c
index 21032b9ba2..5fb6015a90 100644
--- a/providers/fips/fipsprov.c
+++ b/providers/fips/fipsprov.c
@@ -8,6 +8,7 @@
  */
 
 #include <assert.h>
+#include <openssl/configuration.h>
 #include <openssl/core_dispatch.h>
 #include <openssl/core_names.h>
 #include <openssl/params.h>
diff --git a/providers/implementations/rands/seed_src_jitter.c b/providers/implementations/rands/seed_src_jitter.c
index ac57f1c14b..fc64f81241 100644
--- a/providers/implementations/rands/seed_src_jitter.c
+++ b/providers/implementations/rands/seed_src_jitter.c
@@ -9,6 +9,7 @@
 
 #include <string.h>
 #include <openssl/rand.h>
+#include <openssl/configuration.h>
 #include <openssl/core_dispatch.h>
 #include <openssl/e_os2.h>
 #include <openssl/params.h>
-- 
2.43.0

